driver/src/contracts/stablex_auction_element.rs

```diff
@@ -0,0 +1,299 @@
+        );
+
+        let buy_token = u16::from_le_bytes([bytes[53], bytes[52]]);
+        let sell_token = u16::from_le_bytes([bytes[55], bytes[54]]);
+        let valid_from = U256::from(u32::from_le_bytes([
+            bytes[59], bytes[58], bytes[57], bytes[56],
+        ]));
+        let valid_until = U256::from(u32::from_le_bytes([
+            bytes[63], bytes[62], bytes[61], bytes[60],
+        ]));
+        let is_sell_order = bytes[64] > 0;
+        let numerator = BigEndian::read_u128(&bytes[65..81]);
+        let denominator = BigEndian::read_u128(&bytes[81..97]);
+        let amount = BigEndian::read_u128(&bytes[97..113]);
+        let (buy_amount, sell_amount) =
+            compute_buy_sell_amounts(numerator, denominator, amount, is_sell_order);
+        let order_counter = order_count.entry(account_id).or_insert(0);
+        *order_counter += 1;
+        StableXAuctionElement {
+            valid_from,
+            valid_until,
+            sell_token_balance,
+            order: Order {
+                batch_information: Some(BatchInformation {
+                    slot_index: *order_counter - 1,
+                    slot: U256::from(0),
+                }),
+                account_id,
+                buy_token,
+                sell_token,
+                buy_amount,
+                sell_amount,
+            },
+        }
+    }
+}
+
+fn compute_buy_sell_amounts(
+    numerator: u128,
+    denominator: u128,
+    remaining: u128,
+    is_sell_order: bool,
+) -> (u128, u128) {
+    // TODO - use u128_to_u256 for overflow handling (this is a conversion used in naive_solver)
+    assert!(
+        remaining <= denominator,
+        "Smart contract should never allow this inequality"
+    );
+    let mut other = 0 as u128;
+    let (buy_amount, sell_amount) = if is_sell_order {
+        // 0 on sellAmount (remaining <= denominator) is nonsense, but solver can handle it.
+        // 0 on buyAmount (numerator) is a Market Sell Order.
+        if denominator > 0 {
+            // up-casting to handle overflow
+            let top = u128_to_u256(remaining) * u128_to_u256(numerator);
+            let bottom = u128_to_u256(denominator);
+            // Recall that ceil(p / float(q)) == (p + q - 1) / q
+            let other_u256 = (top + bottom - 1) / bottom;
+
+            // Conversion never overflows because remaining <= denominator
+            let x = other_u256.0;
+            other = u128::from(x[0]) + u128::from(x[1]) * (2 as u128).pow(64);
+        }
+        (other, remaining)
+    } else {
+        error!("Unable to to handle buy orders, returning unfulfillable amounts");
+        (0 as u128, 0 as u128)
+        //        if numerator > 0 {
+        //            let p = denominator * remaining;
+        //            let q = numerator;
+        //            other = (p + q - 1) / q;
+        //            (remaining, other)
+        //        }
+    };
+    (buy_amount, sell_amount)
```

Comment:
nit: This line is unnecessary if you remove the `;` after the if statement and don't assign into the variable.

Reply:
nice.